File: HW2/plane.cpp
//------------------------------------------------------------------------------
// plane.cpp - содержит функции обработки самолета
//------------------------------------------------------------------------------

#include "plane.h"
#include <cmath>

//------------------------------------------------------------------------------
// Подсчет рандомного действительного значения
double randomP(double b, double e)
{
    return e + fabs(Random() - b) * 100;
}

//------------------------------------------------------------------------------
// Ввод параметров самолета из потока
void Plane::In(std::ifstream &ifst) {
    ifst >> this->distance_fly >> this->carrying_capacity >> this->speed >> this->range;
}

// Случайный ввод параметров самолета
void Plane::InRnd() {
    this->range = randomP(1.54, 100);
    this->speed = Random();
    this->carrying_capacity = Random();
    this->distance_fly = Random();
}

//------------------------------------------------------------------------------
// Вывод параметров самолета в поток
void Plane::Out(std::ofstream &ofst) {
    ofst << "Plane specifications: distance fly = "
         << this->distance_fly << ", carrying capacity = " << this->carrying_capacity
         << ", Speed = " << this->speed
         << ", Range = " << this->range
         << ". Ideal time of passage = " << this->CountTime() << "\n";
}


//------------------------------------------------------------------------------
// Идеальное время прохождения пути для самолета

double Plane::CountTime() {
    double time = this->range / (double)this->speed;
    return time;
}

File: HW3/src/plane.py
import newRandom


# ----------------------------------------------
# Класс самолета
class Plane:
    # Конструктор самолета
    def __init__(self):
        self.range = 0
        self.speed = 0
        self.distance_fly = 0
        self.carrying_capacity = 0

    # Считывание параметров самолета из строки
    def ReadStrArray(self, strArray, i):
        if i >= len(strArray) - 1:
            return 0
        self.distance_fly = int(strArray[i])
        self.carrying_capacity = int(strArray[i + 1])
        self.speed = int(strArray[i + 2])
        self.range = int(strArray[i + 3])
        i += 4
        return i

    # Случайная генерация параметров самолета
    def RandomIn(self):
        self.range = newRandom.randDouble(10, 10000)
        self.speed = newRandom.randInt(100, 10000)
        self.distance_fly = newRandom.randInt(1000, 1e9)
        self.carrying_capacity = newRandom.randInt(10, 10000)

    # Вывод данных самолета в файл
    def Write(self, ostream):
        ostream.write(
            "Plane: distance_fly = {}, carrying_capacity = {}, speed = {}, range = {}, Ideal time of travel = {}"
            .format(self.distance_fly, self.carrying_capacity, self.speed, self.range, self.CountTime()))
        pass

    #  Идеальное время прохождения пути для самотела
    def CountTime(self):
        return self.range / self.speed
